pkg/util/instancemetadata/prod.go

```diff
@@ -79,7 +82,55 @@ func (p *prod) populateTenantIDFromMSI(ctx context.Context) error {
 	return nil
 }
 
+// populateInstanceMetadataFromEnvironment attempts to populate instance metadata from environment, it does not throw errors.
+func (p *prod) populateInstanceMetadataFromEnvironment() bool {
+	environment := azure.PublicCloud
+	if value, found := os.LookupEnv("AZURE_ENVIRONMENT"); found {
+		var err error
+		environment, err = azure.EnvironmentFromName(value)
+		if err == nil {
+			p.environment = &environment
+		}
+	}
+
+	// Getenv returns empty string if not set which indicates no value for the struct
+	p.subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	p.tenantID = os.Getenv("AZURE_TENANT_ID")
+	p.location = os.Getenv("LOCATION")
+	p.resourceGroup = os.Getenv("RESOURCEGROUP")
+	p.hostname = os.Getenv("HOSTNAME_OVERRIDE")
+
+	if p.hostname == "" {
+		hostname, err := os.Hostname()
+		if err == nil {
+			p.hostname = hostname
+		}
+	}
+
+	return p.haveIntanceMetadata()
+}
+
+func (p *prod) haveIntanceMetadata() bool {
+	return p.environment != nil &&
+		p.subscriptionID != "" &&
+		p.tenantID != "" &&
+		p.location != "" &&
+		p.resourceGroup != "" &&
+		p.hostname != ""
+}
+
 func (p *prod) populateInstanceMetadata() error {
+	if os.Getenv("AZURE_EV2") != "" {
+		if p.populateInstanceMetadataFromEnvironment() {
+			// everything was populated from environment, can stop.
+			p.log.Info("created InstanceMetadata from Environment, not using http metadata endpoint")
+			return nil
```

Comment:
With this change we still making a call to `populateTenantIDFromMSI` to get `tenantID` whichg is not necessary as we already have it from env.